types/field_type.go

```diff
@@ -35,11 +35,15 @@ type FieldType = ast.FieldType
 // NewFieldType returns a FieldType,
 // with a type and other information about field type.
 func NewFieldType(tp byte) *FieldType {
-	return &FieldType{
+	ft := &FieldType{
 		Tp:      tp,
 		Flen:    UnspecifiedLength,
 		Decimal: UnspecifiedLength,
 	}
+	if tp != mysql.TypeVarchar && tp != mysql.TypeVarString && tp != mysql.TypeString {
```

Comment:
Please add a TODO here for the default charset/collation of string-related types.

Reply:
Already set the default collation for string-related types.